### DISCUSSION TITLE
Update Git Repo when app not running

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#705] [Feature Requests]
There is a use case where a Note is made and GitJournal is immediately closed. In that case the changes are not pushed. Perhaps it makes sense to periodically push and pull the changes.

I'd love some feedback on this.

https://pub.dev/packages/workmanager

---
> Please vote on this issue with a 👍 reaction. It's one of the primary mechanisms used to prioritize issues. Please also subscribe to the issue if you're interested, as that way I can discuss possible implementations and share mockups.

**Comment by alensiljak** [comment]
Perhaps a setting for when to synchronise? On start, after change, on exit, manually.

**Comment by Dieterbe** [comment]
> Perhaps a setting for when to synchronise? On start, after change, on exit, manually.

Seems to me there doesn't need to be config options for this. syncing at start and after any change has been made seems logical. Plus the manual button to pull in changes on-demand  (as we have now) seems fine to me.
(in an ideal world, whenever the git repo changes, it would have a way to push the update to gitjournal or notify gitjournal that it should pull, but this is rather hard to achieve)

For "synchronize on exit" to work correctly, it would mean that either:
1) you cannot actually close the app until the push has succeeded.
2) you "close the app" (at least visually) but it leaves something running in the background to complete the sync. (which is what @vHanda suggests I think)

Option 2 seems fair but personally it's not a big deal to me because I can just leave the app running.
